homeassistant/components/filter/sensor.py

```diff
@@ -197,7 +197,15 @@ def _update_filter_sensor_state_event(self, event):
     @callback
     def _update_filter_sensor_state(self, new_state, update_ha=True):
         """Process device state changes."""
-        if new_state is None or new_state.state in [STATE_UNKNOWN, STATE_UNAVAILABLE]:
+        if new_state is None:
+            _LOGGER.warning(
+                "While updating filter %s, the new_state is None.", self._name
+            )
+            self._state = STATE_UNKNOWN
```

Comment:
We never return `STATE_UNKNOWN` bare, instead, return `None`.

```suggestion
            self._state = None
```

Reply:
Can you explain me why considering: 

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L321

Reply:
That is handled on a lower level indeed. For integrations, we return "None" if there is no state to report. We leave it up to the core to handle the `None` state (which it currently transforms it into `STATE_UNKNOWN`.

Reply:
This comment was marked as resolved, however, the code hasn't changed. Unresolving it, as it needs to return None.

Reply:
Sorry, missed that while acknowledging your comment from the mobile app.